src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -798,12 +794,10 @@ private static String getImportedTypeCanonicalName(DetailAST importAst) {
         DetailAST toVisitAst = currentNode.getFirstChild();
         while (toVisitAst == null) {
             toVisitAst = currentNode.getNextSibling();
-            if (toVisitAst == null) {
```

Comment:
The if statement is pointless and so it is safe for it to be removed. Nothing inside the if manipulated `toVisit` which is also the result of this method and nothing inside the method impacts the loop or the result enough that they matter if they are executed or not, even if `toVisit` is not `null` which will end the loop anyways.